src/lib/utils.ts

```diff
@@ -0,0 +1,48 @@
+import * as fs from 'fs';
+import Logger from '../Logger';
+
+/**
+ * Utility helper methods.
+ */
+export class Utils {
+
+  /**
+   * It returns origin chainid by searching through the mosaic config.
+   * @param {number} auxChainId
+   * @param {string} mosaicConfigPath
+   * @returns {number}
+   */
+  public static getOriginChainId(auxChainId: number, mosaicConfigPath: string): number {
+
+    const mosaicConfig = Utils.parseFile(mosaicConfigPath);
+    const auxChainNames = Object.keys(mosaicConfig['auxiliaryChains']);
+    let auxChainIdPresent: boolean = false;
+    let originChainId: number;
+    let chainName: any;
+    for(let i = 0; i < auxChainNames.length && !auxChainIdPresent; i++) {
+      chainName = auxChainNames[i];
+      if(mosaicConfig['auxiliaryChains'][chainName].chainId == auxChainId) {
+        originChainId = mosaicConfig['originChain'].chain;
+        auxChainIdPresent = true;
```

Comment:
should this now break from the `for` loop?

Reply:
Yes, it would break from `for` loop.